app/lib/linters/crunch_api/report_retriever.rb

```diff
@@ -0,0 +1,27 @@
+module Linters
+  module CrunchApi
+
+    # Use the created API ID to get the linting results from 42Crunch using their
+    # API V1 "Retrieve and visualize a security audit report" spec
+    class ReportRetriever
+      def initialize(rest_client: RestClient, base_url: CRUNCH_BASE_URL)
+        @rest_client = rest_client
+        @base_url = base_url
+      end
+
+      def retrieve_report_for_api(api_id)
+        url = URI.join(base_url, "/api/v1/apis/", "#{api_id}/", "assessmentreport").to_s
```

Comment:
Just done a bit of testing with this in the console, it looks like it has to be relative and requires a slash in order to join it together, otherwise it silently fails and doesn't include the api_id

Reply:
Sorry - you are right (see [this](https://apidock.com/ruby/URI/join/class#1245-Be-careful-with-path-vs-endpoint))

I must admit to usually using `File.join` as it's simpler.